### PR TITLE
Tidy should not check line lengths in tests

### LINKED ISSUE #77548: Tidy should not check line lengths in UI tests
UI tests regularly have long lines due to the `//~ ERROR` annotations. So it is not very helpful for tidy to complain about overly long lines... every single time that happened, I ended up adding `// ignore-tidy-linelength` (the amount of time it took me to figure out the exact way to do this went down from 3-5min the first time to <20s by now). To make test maintenance even more annoying, when an error message changes to become shorter, tidy will complain about the ignore flag being unnecessary! And each time the flag is added or removed, you need to re-bless the tests as line numbers change.

This chore doesn't really seem worth it. We don't `rustfmt` the test suite either. Can we just disable the line length checker in the test suite please?

### PR DESCRIPTION
**Pull request by Anthuang**
Tidy will not check line lengths in tests even without the `// ignore-tidy-linelength` annotations. This PR also removes all the annotations which are now unnecessary.

Closes: #77548

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
This looks great, it looks like it might've accidentally deleted a file though? `src/test/ui/raw-str.rs` in particular.

**Comment by petrochenkov** [comment]
I responded to this suggestion previously in https://github.com/rust-lang/rust/issues/75987#issuecomment-682134578.
I think applying tidy to test code has the same value as applying it to any other code.

If the rule is relaxed, the the relaxation should at least be applied only to lines with `//~ XXX` annotations.
To detect such annotations we have regex here - https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/runtest.rs#L3393-L3396.

**Comment by Mark-Simulacrum** [comment]
Note that we're just silencing the line length warning. That's silenced in most of our code anyway due to rustfmt "taking care of it". I am okay with limiting this to just the `//~` lines though, that seems reasonable.

I think there's not too much value in truncating those to 100 characters (unlike regular comments/code).

cc @RalfJung

**Comment by jyn514** [comment]
Please also exclude `// @has` and `// @!has` lines, those come up all the time for rustdoc.

**Comment by Anthuang** [comment]
I've made it check only for lines containing the text you've mentioned, but there are many other lines that are too long (e.g. `// gdbr-check`, `// lldbg-check`, `// CHECK`, `// @matches`, etc.). Should I ignore these too? Or is it better to just leave the `// ignore-tidy-linelength` in the file?

I noticed that some of the lines are long intentionally, such as the line in src/test/ui/slowparse-string.rs, so we may have to leave the `// ignore-tidy-linelength` in some of the files.

**Comment by Mark-Simulacrum** [comment]
I think we want to allow any directives that contain error substrings (i.e., things that cannot be wrapped) via tidy changes, and any tests that remain after that should indeed retain ignore-tidy-linelength.

File: src/tools/tidy/src/style.rs [hunk 1]
```diff
@@ -39,6 +39,24 @@ C++ code used llvm_unreachable, which triggers undefined behavior
 when executed when assertions are disabled.
 Use llvm::report_fatal_error for increased robustness.";
 
+const ANNOTATIONS_TO_IGNORE: &[&str] = &[
+    "// @ has",
```
Comment (jyn514) [inline src/tools/tidy/src/style.rs:43]: I would rather not ignore this; I use this as 'I mean to add this test at some point but I don't want to make it a separate file', but others might not. `htmldocck` doesn't treat it specially.

File: src/tools/tidy/src/style.rs [hunk 2]
```diff
@@ -176,6 +194,8 @@ pub fn check(path: &Path, bad: &mut bool) {
                     a.ends_with("library/std/src/sys/cloudabi") ||
                     a.ends_with("src/doc/book")
             });
+        // let in_tests =
+        //     filename.ends_with(".rs") && file.ancestors().any(|a| a.ends_with("src/test"));
```
Comment (jyn514) [inline src/tools/tidy/src/style.rs:198]:
```suggestion
```

**Comment by Mark-Simulacrum** [comment]
I would prefer that we use the regex in compiletest in addition to the ANNOTATIONS_TO_IGNORE list; it looks like the regex covers more cases than just those enumerated. Please also add a comment to both locations pointing at the other location to make sure they get updated mutually as needed.

Also, it would be great if you could [rebase](https://rustc-dev-guide.rust-lang.org/git.html#advanced-rebasing) your changes such that you have a single commit that changes code and another commit updating the tests for the fallout from that change (i.e., removing linelength annotations from a subset of the tests). If you don't know how to do this, I can try to explain, or I can do it for you, no worries there.

It also looks like CI is failing due to a lack of re-blessing, but I would hold off on doing so for now since it's likely I'll have further feedback after this round of review :)

**Comment by Anthuang** [comment]
Could you point me to where you're talking about? I found this line https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/runtest.rs#L3396 but not sure if this was what you were referring to.

I should be able to manage rebasing!

**Comment by Mark-Simulacrum** [comment]
Yes, sorry, I should have linked to it, I meant https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/runtest.rs#L3393-L3396 (same as you linked). I think I had implicitly linked it in my mind since I was reading @petrochenkov's comment earlier :)

**Comment by Anthuang** [comment]
@rustbot modify labels: +S-waiting-on-review -S-waiting-on-author

Going to change the label without rebasing, since Mark said I can hold off on rebasing.

**Comment by Mark-Simulacrum** [comment]
r=me with a rebase (CI needs fixing, but I think that's part of the rebase in some sense)

**Comment by Anthuang** [comment]
I've just rebased! @Mark-Simulacrum 

@rustbot modify labels: +S-waiting-on-review -S-waiting-on-author

**Comment by RalfJung** [comment]
@Anthuang looks like CI is failing? Lots of diffs like
```
diff of stderr:

1	error[E0161]: cannot move a value of type [u8]: the size of [u8] cannot be statically determined
-	  --> $DIR/return-unsized-from-trait-method.rs:11:17
+	  --> $DIR/return-unsized-from-trait-method.rs:9:17
3	   |
4	LL |         let _ = f.foo();
5	   |                 ^^^^^^^
```

Seems like you need to run the UI test suite with `--bless`.

**Comment by jyn514** [comment]
```
thread '[mir-opt] mir-opt/retag.rs' panicked at 'Actual MIR output differs from expected MIR output /checkout/src/test/mir-opt/retag.{impl#0}-foo.SimplifyCfg-elaborate-drops.after.mir', src/tools/compiletest/src/runtest.rs:3272:25


failures:
    [mir-opt] mir-opt/early_otherwise_branch_68867.rs
    [mir-opt] mir-opt/retag.rs

test result: FAILED. 139 passed; 2 failed; 3 ignored; 0 measured; 0 filtered out
```

**Comment by Anthuang** [comment]
Just fixed the tests, should've done that first sorry

@rustbot modify labels: +S-waiting-on-review -S-waiting-on-author

**Comment by Mark-Simulacrum** [comment]
Please also re-squash commits (the two commit version is ideal, one commit is fine if that's too much hassle).

**Comment by Mark-Simulacrum** [comment]
It looks like there's some unintentional Cargo.lock changes?

**Comment by jyn514** [comment]
```
tidy error: /checkout/src/test/rustdoc/raw-ident-eliminate-r-hashtag.rs: ignoring line length unnecessarily
```

**Comment by Dylan-DPC** [comment]
failed in rollup as well 

@bors r-

**Comment by Mark-Simulacrum** [comment]
@Anthuang It looks like the tests in question failed because we're hard coding line numbers into them -- I imagine you can bump those back by one?

```
    [codegen] codegen\async-fn-debug-msvc.rs
    [codegen] codegen\generator-debug-msvc.rs
```

**Comment by Mark-Simulacrum** [comment]
```
 tidy error: /checkout/src/test/ui/associated-type-bounds/ambiguous-associated-type2.rs: ignoring line length unnecessarily
tidy error: /checkout/src/test/ui/simd-type-generic-monomorphisation.rs: ignoring line length unnecessarily
tidy error: /checkout/src/test/ui/simd-type.rs: ignoring line length unnecessarily
tidy error: /checkout/src/test/codegen/simd-intrinsic/simd-intrinsic-transmute-array.rs: ignoring line length unnecessarily
tidy error: /checkout/src/test/rustdoc/inline_cross/impl_trait.rs: ignoring line length unnecessarily
```

@bors p=5 for next approval

Let me know if you'd like me to fix these on the next go around instead of the back and forth or feel free to ping me on Zulip and I can try to fast-track an r+ for you :)

**Comment by jyn514** [comment]
@Mark-Simulacrum does it make sense to disable the "ignoring length unnecessarily" lint temporarily so this can land? Then once this lands people will stop adding the ignore and it will be easier to turn the lint back on.

**Comment by Mark-Simulacrum** [comment]
Yeah, that'd be fine with me as well, we can definitely do this in two parts.

**Comment by crlf0710** [comment]
@Anthuang  Ping from triage: any updates on this?

**Comment by jyn514** [comment]
@crlf0710 see https://github.com/rust-lang/rust/pull/77675#issuecomment-736571660 - this fails every time it's rolled up or takes more than ~an hour to be approved because almost every single other PR has semantic conflicts. It needs either a `treeclosed` or to disable the lint altogether temporarily (https://github.com/rust-lang/rust/pull/77675#issuecomment-736585939 in order to be merged.

**Comment by JohnCSimon** [comment]
@Anthuang Ping from triage: unfortunately I'm closing this this due to inactivity - please feel free to reopen when you're ready to continue with this.

**Pull request closed without merging**